hardware_interface/src/resource_manager.cpp

```diff
@@ -1283,6 +1305,70 @@ HardwareReadWriteStatus ResourceManager::read(
   return read_write_status;
 }
 
+std::vector<control_msgs::msg::SingleInterfaceValue>
+ResourceManager::get_all_state_interface_values() const
+{
+  std::vector<control_msgs::msg::SingleInterfaceValue> state_interface_values;
+  for (const auto & state_interface_name : resource_storage_->available_state_interfaces_)
+  {
+    try
+    {
+      const auto state_interface_value =
+        resource_storage_->state_interface_map_.at(state_interface_name).get_value();
+      control_msgs::msg::SingleInterfaceValue interface_value;
+      interface_value.header.stamp = resource_storage_->clock_interface_->get_clock()->now();
+      interface_value.interface_name = state_interface_name;
+      interface_value.value = state_interface_value;
+      state_interface_values.push_back(interface_value);
+    }
+    catch (const std::out_of_range & e)
+    {
+      // TODO(Manuel) what to do? should probably never happen but if, is this the right place to
+      // fail?
+    }
+  }
+  return state_interface_values;
+}
+
+void ResourceManager::publish_all_state_interface_values() const
```

Comment:
Why do we have this method? Other method publishes everything and this is what we want. To have one publisher for everything.